packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -154,28 +181,41 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         const xScale = (chart as any).scales['time-axis'];
         const ticks: number[] = xScale.ticks;
         if (ctx && this.props.selectionRange) {
-            const valueStart = this.findNearestValue(this.props.selectionRange.getstart(), ticks);
-            const valueEnd = this.findNearestValue(this.props.selectionRange.getEnd(), ticks);
-            const pixelStart = xScale.getPixelForValue(this.props.selectionRange.getstart(), valueStart);
-            const pixelEnd = xScale.getPixelForValue(this.props.selectionRange.getEnd(), valueEnd);
+            const min = Math.min(this.props.selectionRange.getstart(), this.props.selectionRange.getEnd());
+            const max = Math.max(this.props.selectionRange.getstart(), this.props.selectionRange.getEnd());
+            // If the selection is out of range
+            if (min > this.props.viewRange.getEnd() || max < this.props.viewRange.getstart()) {
+                return;
+            }
+            const minValue = this.findNearestValue(min, ticks);
+            const minPixel = xScale.getPixelForValue(min, minValue);
+            const maxValue = this.findNearestValue(max, ticks);
+            let maxPixel = xScale.getPixelForValue(max, maxValue);
+            // In the case the selection is going out of bounds, the pixelValue needs to be in the displayed range.
+            if (maxPixel === 0) {
+                maxPixel = chart.chartArea.right;
+            }
```

Comment:
Just asking, why do you have to change this method? The behavior of drawing the the selection in XY chart is correct in master. You added the possibility to also select from the XY chart. I'm just wondering why this needs to be modified too.

Reply:
if your selection is partially out of range, the selection is not correctly displayed. Also the borders shown were not the same as for the timeline chart. When I corrected these issues, I had to account for other things as well, start > end and the window out of range completely.